### ISSUE TITLE
Data Lake Option/Consistency

### ISSUE THREAD

**Issue by marshall-sg** [netmeld/netmeld#7]
**Is your feature request related to a problem? Please describe.**
An issue arises around how to store data collected and processed.  Current measures only work in *simple* cases and generally not helpful in the case where large amounts of data is collected or over time.  Basically, source data management is a hybrid of end user and tool (e.g., fetchers) decision points.  It would be helpful to have tools which allow for a more unified source data management option.

**Describe the solution you'd like**
Initial thoughts:
- Tools to migrate/store raw data to a consistent location with some ordering.
  - Hopefully the consistent location enables end user search and discovery.
  - Thought might want to be put into tools (i.e., `import-*`) leveraging the layout to allow auto-loading of data.
  - What is *best* is debatable, but useful alternatives may leverage linking to achieve allow easier retrieval for tools or end users. Location will probably consist of things like `host`, `import_tool`, `collection_method`, etc.
- The storing location is some sort of version controlled repository.
  - This hopefully loosens the need to `DTS_UUID` everything as that may be handled by the versioning within the repo.
  - Ideally you would be able to load data from selected points in time so you could compare state/knowledge of the targeted environment over the course of time.
- Ideally, the ability to permanently purge or alter data.
  - Basically, if a mistake is made it needs to be able to be *erased*.  That is suppose I call a host by the wrong name, I need to be able to correct it for the life (backwards) of the repo as the data is still valid but stored under an incorrect name.  Similarly, if data for a target that is later identified to be out-of-scope is stashed, that data would need to be able to be purged as if it never existed.
- Tooling should be generic enough to allow different back ends.
  - While one solution may be the default/supported, allow for end users to create the logic to support different backends.

**Additional context**
This should probably be another module as an initial prototype is wanted to facilitate discussion points about what works and what doesn't.
